### ISSUE TITLE
Roadmap to v1.0.0

### ISSUE THREAD

**Issue by clue** [reactphp/child-process#31]
Let's face it, this project is currently *beta*, but has been used in production for years :shipit: 

We're currently following a v0.X.Y release scheme (http://sentimentalversioning.org/).

We should finally make this *stable* and fully adhere to SemVer and release a stable v1.0.0.

To a large extend, a stable v1.0.0 helps making BC breaks more explicit and thus the whole project more reliable from a consumer perspective. This project is actively maintained and has received some major updates in the last weeks and has some major updates planned in the next weeks. Given our current versioning scheme, we'd like to ensure all anticipated BC breaks will be merged before the planned v1.0.0 release.

As such, I've set up a roadmap that enlists only the major changes for each version among with planned release dates towards a stable v1.0.0 release:

## v0.4.0 :white_check_mark: 

* Released 2014-07-31
* Initial tagged version

## v0.4.1 :white_check_mark: 

* Released 2016-08-01
* Standalone component and tests

## v0.4.2 :white_check_mark: 

* Released 2017-03-10
* Stream v0.5 API

## v0.4.3 :white_check_mark: 

* Released 2017-03-14
* Support PHP 5.3 - PHP 7.1

## v0.5.0 :white_check_mark: 

* Released 2017-08-15
* Readable/Writable pipes
* Drop Windows support entirely

## v0.6.0 :white_check_mark: 

* Releases 2019-01-14
* Custom pipes
* Add (limited) Windows support

## v0.7.0

* Planned 2025-Q1??
* API cleanup

## v1.0.0

* Planned 2025-??
* No new changes planned, this should merely mark the previous release as "stable"

This ticket aims to serve as a basic overview and does not contain every single change. Please also see the milestone links and the CHANGELOG for more details.

Obviously, this roadmap is subject to change and I'll try to keep it updated as we progress. In order to avoid cluttering this, please keep discussion in this ticket to a minimum and consider reaching out to us through new tickets or Twitter etc.

**Comment by clue** [comment]
Updated now that the v0.4.3 release is out (a few days ago already) :shipit:

**Comment by WyriHaximus** [comment]
Updated now that the v0.5.0 release is out :shipit:

**Comment by inoas** [comment]
My 2 cents are:
- If your intention is to still support PHP 5.x make 1.0.0 support ^PHP 5.6.0. Do not change any code aside some composer requirements. Just set PHP 5.6 as a requirement. This should happen right now.

- Then 2.0 can target PHP 7.0 once that's beneficial and 1.0 can take backports

**Comment by inoas** [comment]
The features for 0.6/0.7 could go into 2.0 as well and that could stay PHP 5.6 compatible and only 3.0 would go for PHP 7

**Comment by clue** [comment]
Updated now that the v0.6.0 release is out :shipit:

**Comment by WyriHaximus** [comment]
@clue is 0.7 still going to be a thing?

**Comment by clue** [comment]
> @clue is 0.7 still going to be a thing?

Good point, but yes, that's still my understanding. I've just filed #72 to discuss this further :+1:
